FBTweak/FBTweak.m

```diff
@@ -179,17 +181,39 @@ - (instancetype)initWithIdentifier:(NSString *)identifier name:(NSString *)name
                       defaultValue:(FBTweakValue)defaultValue {
   if (self = [super initWithIdentifier:identifier name:name defaultValue:defaultValue]) {
     NSData *archivedValue = [[NSUserDefaults standardUserDefaults] objectForKey:identifier];
-    self.currentValue = (archivedValue != nil && [archivedValue isKindOfClass:[NSData class]] ?
-                         [NSKeyedUnarchiver unarchiveObjectWithData:archivedValue] : archivedValue);
+    if (archivedValue != nil && [archivedValue isKindOfClass:[NSData class]]) {
+      self.currentValue =
+          [NSKeyedUnarchiver unarchivedObjectOfClasses:[FBPersistentTweak validValueClasses]
+                                              fromData:archivedValue error:nil];
+    } else {
+      self.currentValue = archivedValue;
+    }
   }
 
   return self;
 }
 
 - (void)setCurrentValue:(FBTweakValue)currentValue {
   [super setCurrentValue:currentValue];
+
   // we can't store UIColor to the plist file. That is why we archive value to the NSData.
-  [[NSUserDefaults standardUserDefaults] setObject:[NSKeyedArchiver archivedDataWithRootObject:currentValue] forKey:self.identifier];
+  NSError *error;
+  NSData *data = [NSKeyedArchiver archivedDataWithRootObject:currentValue requiringSecureCoding:YES
+                                                   error:&error];
+  NSAssert(data != nil, @"Failed to archive value with error: %@", error);
+  [[NSUserDefaults standardUserDefaults] setObject:data forKey:self.identifier];
+}
+
++ (NSSet<Class> *)validValueClasses {
+  return [NSSet setWithArray:@[
+    [NSData class],
+    [NSString class],
+    [NSNumber class],
+    [NSDate class],
+    [NSArray class],
```

Comment:
are array and dict really valid here? (also they will be unarchived regardless).

Reply:
According to documentation of `FBTweakValue` they are valid (the only restriction is that it should be persisted in user defaults).